src/vivarium_census_prl_synth_pop/results_processing/ssn_and_itin.py

```diff
@@ -254,10 +246,11 @@ def do_collide_ssns(
         & (obs_data["employer_id"] != -1)
         & (~obs_data["has_ssn"])
     )
+    # TODO: Replace with sampling from the artifact data if we figure out a
+    # more performant way of doing so (eg querying the hdf with randomized idx)
```

Comment:
This should be easy? Just generate a N random ints in the range [0:len(ssns_in_artifact)] and store them as an index. Then look up those indices in the artifact.

```
ssn_idx = pd.Index(
    random_integers(
        min_val=0,
        max_val=num_possible_ssns,
        index=pd.Index(range(create_ssn_mask.sum()),
        randomness=randomness,
        additional_key=f"{seed}_collide",
    )
)
obs_data.loc[create_ssn_mask, "ssn"] = ssns_from_artifact.loc[ssn_idx]
```

Reply:
I don't think this solves the problem because it's still loading all ssns_from_artifact which is what takes a long time. That said, I don't think this takes all that long and isn't a big deal anyway